pkgs/watcher/test/directory_watcher/file_tests.dart

```diff
@@ -545,4 +545,132 @@ void _fileTests({required bool isNative}) {
       isRemoveEvent('some_name.txt')
     ]);
   });
+
+  // Watching a relative path is not a great idea because the meaning of the
+  // path changes if `Directory.current` changes, leading to surprising and
+  // undefined behavior. But released `package:watcher` allows it so at least
+  // check basic functionality works.
+  test('watch relative directory', () async {
+    // Testing with relative paths is hard! There's only one current directory
+    // across the whole VM, tests can run in different isolates, and this test
+    // runs twice: once with the native watcher and once with the polling one.
+    // Use the current directory name as a check to ensure both don't run at
+    // the same time. All other tests must use absolute paths.
+    final testDirectory = Directory(p.join(d.sandbox, 'for_relative_test'));
+    testDirectory.createSync();
```

Comment:
[nit] Prefer to use the `test_descriptor` APIs to make changes under the sandbox directory.

Reply:
Done. Well, sort of, these tests already have "createDir" that creates in the sandbox, used that.